extensions/sql-assessment/src/htmlReportGenerator.ts

```diff
@@ -0,0 +1,219 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { LocalizedStrings } from './localized';
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { escape } from './utils';
+
+export class HTMLReportBuilder {
+	constructor(
+		private _assessmentResult: azdata.SqlAssessmentResult,
+		private _dateUpdated: number,
+		private _connectionInfo: azdata.connection.ConnectionProfile
+	) {
+	}
+
+	public async build(): Promise<string> {
+		const serverInfo = await azdata.connection.getServerInfo(this._connectionInfo.connectionId);
+
+		let mainContent = `
+		<html>
+		<head>
+			<title>${LocalizedStrings.REPORT_TITLE}</title>
+		</head>
+		<body>
+			<div class="header">
+				<div>${LocalizedStrings.REPORT_TITLE}</div>
+			</div>
+			<div style="font-style: italic;">${new Date(this._dateUpdated).toLocaleString(vscode.env.language)}</div>
+			${this.buildVersionDetails(serverInfo)}
+			<div style="margin-top: 20px;">
+				${this.buildResultsSection()}
+			</div>
+			${this.buildStyleSection()}
+		</body>
+		</html>`;
+		return mainContent;
+	}
+
+	private instanceName(serverInfo: azdata.ServerInfo): string {
+		const serverName = this._connectionInfo.serverName;
+		if (['local', '(local)', '(local);'].indexOf(serverName.toLowerCase()) >= 0) {
+
+			return serverInfo !== undefined
+				? (<any>serverInfo)['machineName']
+				: serverName;
+		}
+		return serverName;
+	}
+
+	private buildVersionDetails(serverInfo: azdata.ServerInfo): string {
```

Comment:
It seems like this isn't being used?

Reply:
This is needed for the header to display information about the SQL server instance.